Sources/ComposableArchitecture/Internal/Binding+IsPresent.swift

```diff
@@ -0,0 +1,16 @@
+import SwiftUI
+
+extension Binding {
+  func isPresent<Wrapped>() -> Binding<Bool> where Value == Wrapped? {
+    .init(
+      get: { self.wrappedValue != nil },
+      set: { isPresent in
+        if !isPresent {
+          withTransaction(self.transaction) {
+            self.wrappedValue = nil
+          }
```

Comment:
`withTransaction` might not be needed here because `self.transaction` should already be applied to `self.wrappedValue`. It might even override other `withTransaction/withAnimation` calls higher up.
In the event you wanted to add support for `.transaction()/.animation()` to this binding, I've found the best way to maintain the animation precedence would be
```swift
extension Binding {
  func isPresent<Wrapped>() -> Binding<Bool> where Value == Wrapped? {
    .init(
      get: { self.wrappedValue != nil },
      set: { isPresent, transaction in
        guard !isPresent else { return }
        self.transaction(transaction).wrappedValue = nil
      }
    )
  }
}
```

Reply:
Interesting! Good to know, thanks @iampatbrown!